core/src/main/java/org/jdbi/v3/core/statement/HashPrefixSqlParser.java

```diff
@@ -36,23 +34,20 @@
  * <code>#tokenName</code>.
  */
 public class HashPrefixSqlParser implements SqlParser {
-    private final Map<String, ParsedSql> cache = Collections.synchronizedMap(new WeakHashMap<>());
+    private static final JdbiCache<String, ParsedSql> PARSED_SQL_CACHE =
+        JdbiCaches.declare(HashPrefixSqlParser::internalParse);
 
     @Override
     public ParsedSql parse(String sql, StatementContext ctx) {
-        try {
-            return cache.computeIfAbsent(sql, this::internalParse);
-        } catch (IllegalArgumentException e) {
-            throw new UnableToCreateStatementException("Exception parsing for named parameter replacement", e, ctx);
-        }
+        return PARSED_SQL_CACHE.get(sql, ctx);
```

Comment:
The exception thrown by the ANTLR parser is unintuitive for most Jdbi users. Let's continue wrapping any parse exception in a Jdbi exception so parsing exceptions are easier to recognize.